builtin/logical/database/path_roles.go

```diff
@@ -504,17 +540,20 @@ func (b *databaseBackend) pathStaticRoleCreateUpdate(ctx context.Context, req *l
 		if err := req.Storage.Put(ctx, entry); err != nil {
 			return nil, err
 		}
+		item, err = b.popFromRotationQueueByKey(name)
+		if err != nil || item == nil {
+			b.Logger().Warn("expected role to exist in rotation queue but none found", "role", name, "error", err)
+			item = &queue.Item{
+				Key:      name,
+				Priority: lvr.Add(role.StaticAccount.RotationPeriod).Unix(),
+			}
```

Comment:
```suggestion
			return nil, err
```

Reply:
Previously, the API would treat updates on static roles that don't exist mostly like creates, except it wouldn't trigger an immediate password rotation. That's a bug, because it means Vault is tracking a static account that it doesn't know the password for, but it's a separate bug and one that should be exceedingly rare, because we have an existence check on static roles that should ensure we almost always get the correct create/update operation for a static role depending on whether it already exists. I think it might be possible to have a race if there is one delete and one update request in flight at the same time, which is why I say it should be exceedingly rare to encounter this bug.

However, given the size of this PR, and the number of versions we want to backport to, I'd rather not take the risk of changing this bit of API unless we think it's part of the problem.